src/pages/signin.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { ChangeEvent, FormEvent, useState } from "react";
+import Image from "next/image";
+import logoCircle from "../public/circle.png";
+import logo from "../public/logo.png";
+import Input from "@components/Input";
+import CheckboxRadix from "@components/Checkbox/Checkbox";
+import { Button } from "@components/Button";
+import { FcGoogle } from "react-icons/fc";
+import { GrLinkedinOption } from "react-icons/gr";
+import { useMutation, gql } from "@apollo/client";
+import Link from "next/link";
+
+const SIGN_IN_USER = gql`
+  mutation LoginUser($email: String!, $password: String!) {
+    signInUser(userInput: { email: $email, password: $password }) {
+      token
+      user {
+        id
+      }
+    }
+  }
+`;
+
+const SignIn = () => {
+  const [formState, setFormState] = useState({
+    username: "",
+    password: "",
+  });
+
+  const handleStrategyLogin = async (route: string) => {
+    window.location.href = `http://localhost:3030${route}`;
+  };
+
+  const [signInUser, { data, error }] = useMutation(SIGN_IN_USER);
+
+  const handleInputChange = (name: string, value: string) => {
+    setFormState({
+      ...formState,
+      [name]: value,
+    });
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      const { data } = await signInUser({
+        variables: { email: formState.username, password: formState.password },
```

Comment:
Se for fazer a abordagem de usar uncontrolled components é só fazer um Object.entries(formRef.current) que deve funcionar.